// orionsoft-parts/lib/components/DropOver/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _temp;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactAddonsCssTransitionGroup = require('react-addons-css-transition-group');

var _reactAddonsCssTransitionGroup2 = _interopRequireDefault(_reactAddonsCssTransitionGroup);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DropOver = (_temp = _class = function (_React$Component) {
  (0, _inherits3.default)(DropOver, _React$Component);

  function DropOver() {
    (0, _classCallCheck3.default)(this, DropOver);
    return (0, _possibleConstructorReturn3.default)(this, (DropOver.__proto__ || Object.getPrototypeOf(DropOver)).apply(this, arguments));
  }

  (0, _createClass3.default)(DropOver, [{
    key: 'renderMenu',
    value: function renderMenu() {
      if (!this.props.open) return;
      return _react2.default.createElement(
        'div',
        { className: 'os_dropover', key: 'menu' },
        this.props.children
      );
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        _reactAddonsCssTransitionGroup2.default,
        {
          transitionName: 'os_dropover',
          transitionEnterTimeout: 200,
          transitionLeaveTimeout: 200 },
        this.renderMenu()
      );
    }
  }]);
  return DropOver;
}(_react2.default.Component), _class.propTypes = {
  open: _propTypes2.default.bool,
  children: _propTypes2.default.node
}, _temp);
exports.default = DropOver;


//////////////////
// WEBPACK FOOTER
// ./node_modules/orionsoft-parts/lib/components/DropOver/index.js
// module id = 1540
// module chunks = 93
