// orionsoft-parts/lib/components/BounceLoading/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _temp;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var BounceLoading = (_temp = _class = function (_React$Component) {
  (0, _inherits3.default)(BounceLoading, _React$Component);

  function BounceLoading() {
    (0, _classCallCheck3.default)(this, BounceLoading);
    return (0, _possibleConstructorReturn3.default)(this, (BounceLoading.__proto__ || Object.getPrototypeOf(BounceLoading)).apply(this, arguments));
  }

  (0, _createClass3.default)(BounceLoading, [{
    key: 'render',
    value: function render() {
      var itemStyle = {
        height: this.props.size,
        width: this.props.size,
        backgroundColor: this.props.color
      };
      return _react2.default.createElement(
        'div',
        { className: 'os_bounce_loading' },
        _react2.default.createElement('div', { className: 'os_bounce_bounce1', style: itemStyle }),
        _react2.default.createElement('div', { className: 'os_bounce_bounce2', style: itemStyle }),
        _react2.default.createElement('div', { className: 'os_bounce_bounce3', style: itemStyle })
      );
    }
  }]);
  return BounceLoading;
}(_react2.default.Component), _class.propTypes = {
  size: _propTypes2.default.number,
  color: _propTypes2.default.string
}, _class.defaultProps = {
  size: 10,
  color: '#c1c1c1'
}, _temp);
exports.default = BounceLoading;


//////////////////
// WEBPACK FOOTER
// ./node_modules/orionsoft-parts/lib/components/BounceLoading/index.js
// module id = 475
// module chunks = 93
